src/Cool/String/String.cpp

```diff
@@ -192,11 +171,85 @@ auto remove_whitespaces(std::string_view text) -> std::string
     return res;
 }
 
+auto is_commented_out(std::string_view text) -> bool
+{
+    const auto comment_pos    = text.find("//");
+    const auto first_word_pos = text.find_first_not_of("/ ");
+    return comment_pos != std::string_view::npos &&
+           comment_pos < first_word_pos;
+}
+
+template<typename String>
+static auto substring_impl(
+    const String& text,
+    size_t        begin,
+    size_t        end
+) -> String
+{
+    assert(begin != String::npos);
+    assert(end != String::npos);
+    assert(begin <= end);
+    return text.substr(begin, end - begin);
+}
+
+auto substring(
+    const std::string&        text,
+    std::pair<size_t, size_t> begin_end
+) -> std::string
+{
+    return substring_impl(text, begin_end.first, begin_end.second);
+}
+
+auto substring(
+    std::string_view          text,
+    std::pair<size_t, size_t> begin_end
+) -> std::string_view
+{
+    return substring_impl(text, begin_end.first, begin_end.second);
+}
+
+auto substring(
+    const std::string& text,
+    size_t             begin,
+    size_t             end
+) -> std::string
+{
+    return substring_impl(text, begin, end);
+}
+
+auto substring(
+    std::string_view text,
+    size_t           begin,
+    size_t           end
+) -> std::string_view
+{
+    return substring_impl(text, begin, end);
+}
+
+auto find_next_word_position(
+    std::string_view text,
+    size_t           offset,
+    std::string_view delimiters
+) -> std::optional<std::pair<size_t, size_t>>
+{
+    const size_t idx1 = text.find_first_not_of(delimiters, offset);
```

Comment:
Use `auto`. If you don't, I have to go look up the documentation to make sure the return type of `find_first_not_of `is indeed `size_t `and you are not making an implicit conversion.
In this case you don't really care about having a `size_t`, you care about having what `find_first_not_of `returns you.